content/en/blog/_posts/2024-08-16-Honor-PV-Reclaim-Policy-Beta.md

```diff
@@ -0,0 +1,195 @@
+```
+The following output shows the PVC and its bound PV; the PV is shown under the `VOLUME` column:
+```
+NAME                        STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS               AGE
+example-vanilla-block-pvc   Bound    pvc-6791fdd4-5fad-438e-a7fb-16410363e3da   5Gi        RWO            example-vanilla-block-sc   19s
+```
+
+#### Delete PV
+
+When I try to delete a bound PV, the kubectl session blocks and the `kubectl` 
+tool does not return back control to the shell; for example:
+
+```
+kubectl delete pv pvc-6791fdd4-5fad-438e-a7fb-16410363e3da
+```
+
+```
+persistentvolume "pvc-6791fdd4-5fad-438e-a7fb-16410363e3da" deleted
+^C
+```
+
+#### Retrieving the PV:
+```
+kubectl get pv pvc-6791fdd4-5fad-438e-a7fb-16410363e3da
+```
+
+It can be observed that the PV is in a `Terminating` state
+```
+NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS        CLAIM                               STORAGECLASS               REASON   AGE
+pvc-6791fdd4-5fad-438e-a7fb-16410363e3da   5Gi        RWO            Delete           Terminating   default/example-vanilla-block-pvc   example-vanilla-block-sc            2m23s
+```
+
+#### Delete PVC
+
+```
+kubectl delete pvc example-vanilla-block-pvc
+```
+
+The following output is seen if the PVC gets successfully deleted:
+```
+persistentvolumeclaim "example-vanilla-block-pvc" deleted
+```
+
+The PV object from the cluster also gets deleted. When attempting to retrieve the PV
+it will be observed that the PV is no longer found:
+
+```
+kubectl get pv pvc-6791fdd4-5fad-438e-a7fb-16410363e3da
+```
+
+```
+Error from server (NotFound): persistentvolumes "pvc-6791fdd4-5fad-438e-a7fb-16410363e3da" not found
+```
+
+Although the PV is deleted, the underlying storage resource is not deleted and
+needs to be removed manually.
+
+To sum up, the reclaim policy associated with the PersistentVolume is currently
+ignored under certain circumstances. For a `Bound` PV-PVC pair, the ordering of PV-PVC
+deletion determines whether the PV reclaim policy is honored. The reclaim policy
+is honored if the PVC is deleted first; however, if the PV is deleted prior to
+deleting the PVC, then the reclaim policy is not exercised. As a result of this behavior,
+the associated storage asset in the external infrastructure is not removed.
+
+## PV reclaim policy with Kubernetes v1.31
+
+The new behavior ensures that the underlying storage object is deleted from the backend when users attempt to delete a PV manually.
+
+#### How to enable new behavior?
+
+To take advantage of the new behavior, you must have upgraded your cluster to the v1.31 release of Kubernetes
+and run the CSI [`external-provisioner`](https://github.com/kubernetes-csi/external-provisioner) version `5.0.1` or later.
+
+#### How does it work?
+
+For CSI volumes, the new behavior is achieved by adding a finalizer `external-provisioner.volume.kubernetes.io/finalizer`
```

Comment:
Could link to https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/ here.